tests/foreman/cli/test_ansible.py

```diff
@@ -103,3 +104,50 @@ def test_positive_ansible_e2e(target_sat, module_org, rhel_contenthost):
     assert SELECTED_ROLE, (
         SELECTED_VAR not in target_sat.cli.Ansible.variables_info({'name': SELECTED_VAR}).stdout
     )
+
+
+@pytest.mark.tier2
+def test_add_and_remove_ansible_role_hostgroup(target_sat):
+    """
+    Test add and remove functionality for ansible roles in hostgroup via cli
+
+    :id: 2c6fda14-4cd2-490a-b7ef-7a08f8164fad
+
+    :customerscenario: true
+
+    :Steps:
+        1. Create a hostgroup
+        2. Sync few ansible roles
+        3. Assign/Add some ansile roles with the hostgroup
+        4. Remove the added ansible roles from the hostgroup
+
+    :expectedresults:
+        1. Ansible role add/remove functionality should work as expected in cli
+
+    :BZ: 2029402
+
+    :CaseAutomation: Automated
+    """
+    ROLES = [
+        'theforeman.foreman_scap_client',
+        'redhat.satellite.hostgroups',
+        'RedHatInsights.insights-client',
+    ]
+    proxy_id = target_sat.nailgun_smart_proxy.id
+    hg_name = gen_string('alpha')
+    result = target_sat.cli.HostGroup.create({'name': hg_name})
+    assert 'Hostgroup created.' in result[0]['message']
+
+    target_sat.cli.Ansible.roles_sync({'role-names': ROLES, 'proxy-id': proxy_id})
```

Comment:
can we not sync all roles here?

Reply:
Here we don't require to sync all the roles :)